### ISSUE TITLE
1.1.10 panic: unknown connector service: `business_central` crashes on plan

### ISSUE THREAD

**Issue by ann8ty** [fivetran/terraform-provider-fivetran#248]
**Describe the bug**
The business central connector is a preview connector.  I was testing upgrading our TF to the latest version but it crashes.

fivetran = {
      source  = "fivetran/fivetran"
      version = "0.7.3". (works / previous)
      version = "~> 1.1.10" (crashes)

**To Reproduce**

```
resource "fivetran_connector" "business_central_connector" {
  group_id           = data.aws_ssm_parameter.fivetran_group_id_external_id_versioned.value
  service            = "business_central"
  run_setup_tests    = true
  trust_fingerprints = true
  trust_certificates = true

  destination_schema {
    name = local.business_central_connector_name
  }
}
```

**Expected behavior**
Not crash.

**Logs & Output**

plan output:
```
...
fivetran_connector.amplitude_connector: Refreshing state... [id=xya]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-fivetran_v1.1.10 plugin:

panic: unknown connector service: `business_central`

goroutine 15 [running]:
github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model.getDestinationSchemaValuesMap({0x140003b9550, 0x10}, {0x1400011c288, 0x14})
        github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model/connector.go:344 +0x578
github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model.getDestinationSchemaValue({0x140003b9550, 0x10}, {0x1400011c288, 0x14})
        github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model/connector.go:337 +0x150
github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model.(*ConnectorResourceModel).ReadFromContainer(0x1400011e200, {{0x140003b9510, 0xe}, {0x1400011c288, 0x14}, {0x1400011c2b8, 0x15}, {0x14000628300, 0x24}, {0x1400011c240, ...}, ...})
        github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model/connector.go:229 +0x12c
github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model.(*ConnectorResourceModel).ReadFromResponse(_, {{{0x140003b9478, 0x7}, {0x0, 0x0}}, {{{0x140003b9510, 0xe}, {0x1400011c240, 0x11}, {0x140003b9550, ...}, ...}, ...}})
        github.com/fivetran/terraform-provider-fivetran/fivetran/framework/core/model/connector.go:132 +0x84
github.com/fivetran/terraform-provider-fivetran/fivetran/framework/resources.(*connector).Read(0x14000114058, {0x1039761c0, 0x1400088cf00}, {{{{0x10397ae58, 0x14000f6cf30}, {0x10385bb20, 0x140007e6480}}, {0x10397c788, 0x140007b62d0}}, 0x1400028e038, ...}, ...)
        github.com/fivetran/terraform-provider-fivetran/fivetran/framework/resources/connector.go:197 +0x578
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x140000e8000, {0x1039761c0, 0x1400088cf00}, 0x1400088cf90, 0x14000793540)
        github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_readresource.go:101 +0x4a4
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x140000e8000, {0x1039761c0?, 0x1400088cdb0?}, 0x1400060e640)
        github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_readresource.go:55 +0x1a4
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ReadResource(0x103976118?, {0x1039761c0?, 0x1400088ca80?}, 0x1400060e640)
        github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf6muxserver/mux_server_ReadResource.go:35 +0x188
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0x140004cdc20, {0x1039761c0?, 0x1400088c2a0?}, 0x14000099320)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov6/tf6server/server.go:787 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x103938620?, 0x140004cdc20}, {0x1039761c0, 0x1400088c2a0}, 0x140001b2000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:431 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000276000, {0x10397b320, 0x140002949c0}, 0x1400089e000, 0x1400070ac00, 0x103f3bbe8, 0x0)
        google.golang.org/grpc@v1.57.0/server.go:1360 +0xc88
google.golang.org/grpc.(*Server).handleStream(0x14000276000, {0x10397b320, 0x140002949c0}, 0x1400089e000, 0x0)
        google.golang.org/grpc@v1.57.0/server.go:1737 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.57.0/server.go:982 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.57.0/server.go:980 +0x16c

Error: The terraform-provider-fivetran_v1.1.10 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Provide terraform CLI output on `plan` | `apply`. If it is possible, please, refer to [this guide](https://developer.hashicorp.com/terraform/internals/debugging) to enable debug output.

**Plugin version:**
1.1.0

**Additional context**

1.0.0 also fails
```
...
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: read error
│ 
│   with fivetran_connector.business_central_connector,
│   on connector-business-central.tf line 4, in resource "fivetran_connector" "business_central_connector":
│    4: resource "fivetran_connector" "business_central_connector" {
│ 
│ unknown service: `business_central`
```

**Comment by beevital** [comment]
Thank you for this report. Working on it!

**Comment by beevital** [comment]
I found the issue: the problem is that this connector is in Private Preview state. And it's not included in our Open API schema.
We are working on this issue, and the service will be included into OAS with the next release I believe.

Previously we didn't check if service is represented in OAS so it was working for older versions of Terraform.

**Comment by beevital** [comment]
Fix merged, release v1.1.11.

**Comment by ann8ty** [comment]
fixed!
